src/Network/Haskoin/Store/Data.hs

```diff
@@ -590,6 +649,33 @@ data TxData = TxData
     , txDataTime    :: !Word64
     } deriving (Show, Eq, Ord, Generic, Serialize)
 
+instance BinSerial TxData where
+  binSerial _ TxData
+        { txDataBlock   = blockRef
+        , txData        = tx
+        , txDataPrevs   = dataPrev
+        , txDataDeleted = dataDeleted
+        , txDataRBF     = dataRbf
+        , txDataTime    = time
+        } = do
+      put blockRef
+      put tx
+      put dataPrev
+      put dataDeleted
+      put dataRbf
+      putWord64be time
```

Comment:
Use `block_ref` or just `r` here as variable name. In Haskoin Store use `camelCase` notation for identifiers at the top level such as functions and records on data types. For bindings that are scoped to a function and do not escape use short names frome one to three-letters long or longer multiple-word identifiers in `snake_case` format.